pkg/secrethub/client.go

```diff
@@ -102,7 +104,7 @@ func NewClient(with ...ClientOption) (*Client, error) {
 
 	// Try to use default key credentials if none provided explicitly
 	if client.decrypter == nil {
-		err := WithCredentials(credentials.UseKey(nil, nil))(client)
+		err := WithCredentials(credentials.UseKey(client.DefaultCredential()))(client)
```

Comment:
This is a bit of a 'hacky' fix to apply an option after the client has already been initialized, correct?

Should we then create a dedicated (private) `with(options ...Options)` function for this?

Reply:
Hmm, that's an interesting approach.

Reply:
Tried it and works nicely.